JASP-Engine/JASP/R/commonTTest.R

```diff
@@ -18,60 +18,327 @@
 #' The plan of this file is to successively abstract functionality
 #' from the individual t-tests into a common interface to reduce clutter
 #'
-#' returns the result object and the dataset
-.initializeTTest <- function(dataset, options, perform, type = 'independent-samples') {
-	
-	if (type == "one-sample") {
-		analysisTitle <- "One Sample T-Test"
-	} else if (type == "paired") {
-		analysisTitle <- "Paired Samples T-Test"
-	} else {
-		analysisTitle <- "Independent Samples T-Test"
-	}
-	
-	groups <- options$groupingVariable
-	depvars <- unlist(options$variables)
-	
-	## a paired t-test works with pairs ...
-	if (type == 'paired') {
-		depvars <- unlist(options$pairs)
-		depvars <- depvars[depvars != ""]
-	}
-	
-	## !is.null(groups) is there to avoid errors for paired and one-sample t-test
-	if (!is.null(groups) && groups == "") groups <- NULL
+.ttestReadData <- function(dataset, options, type) {
+  if (!is.null(dataset))
+    return(dataset)
+  else {
+    groups  <- options$groupingVariable
+    if (!is.null(groups) && groups == "") 
+      groups <- NULL
+    if(type %in% c("one-sample", "independent"))
+      depvars <- unlist(options$variables)
+    else if (type == 'paired') {
+      depvars <- unlist(options$pairs)
+      depvars <- depvars[depvars != ""]
+    }
+    exclude <- NULL
+    if (options$missingValues == "excludeListwise")
+      exclude <- depvars
+    return(.readDataSetToEnd(columns.as.numeric  = depvars,
+                             columns.as.factor   = groups,
+                             exclude.na.listwise = exclude))
+  }
+}
+
+.ttestCheckErrors <- function(dataset, options, type) {
+  if(type == "paired") {
+    for (pair in options$pairs) {
+      if(pair[[1]] == "" || pair[[2]] == "")
+        next
+      p1 <- pair[[1]]
+      p2 <- pair[[2]]
+      if(is.null(p1) || is.null(p2))
+        return()
+      datasetErrorCheck <- data.frame(dataset[[.v(p1)]] - dataset[[.v(p2)]])
+      colnames(datasetErrorCheck) <- .v(paste0("Difference between ", p1, " and ", p2))
+      .hasErrors(datasetErrorCheck, 
+                 type = "variance", 
+                 exitAnalysisIfErrors = TRUE)
+      .hasErrors(dataset,
+                 type = c('observations', 'variance', 'infinity'),
```

Comment:
Maybe my comment in https://github.com/jasp-stats/jasp-desktop/pull/3560#discussion_r331474754 was not very clear. If you perform these error checks here, then the analysis won't show everything that can still be computed.

And you get this difference, between how it used to be and how you changed it:
New <-> Old
<img width="2536" alt="Screenshot 2019-11-05 at 12 13 25" src="https://user-images.githubusercontent.com/18737241/68203268-b6154c00-ffc5-11e9-8c21-0a388cc15b37.png">

Do you think you can make the new sitation as robust as the old one? Otherwise I can also do it at a later otherwise